walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -19,36 +19,58 @@
 import java.util.Collections;
 import java.util.List;
 
-/** Finds possible meeting times for a request for all mandatory participants */
+/** Finds possible meeting times for a request for all mandatory participants and potentially optional participants*/
 public final class FindMeetingQuery {
   /**
    * @param events List of all known events with times and participants
    * @param request Meeting to be planned
    * @return List of all possible meeting times
    */
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    List<TimeRange> busyTimes = getBusyTimes(events, request);
-    Collections.sort(busyTimes, TimeRange.ORDER_BY_START);
-    busyTimes = mergeIntervals(busyTimes);
-    return getAvailableTimes(busyTimes, request.getDuration());
+    List<TimeRange> mandatoryBusyTimes = getBusyTimes(events, request, "mandatory", Collections.emptyList());
+    List<TimeRange> mandatoryResult = getAvailableTimes(mandatoryBusyTimes, request.getDuration());
+
+    // If there are optional attendees, account for them in schedule
+    if (!request.getOptionalAttendees().isEmpty()) {
+      List<TimeRange> optionalBusyTimes = getBusyTimes(events, request, "optional", mandatoryBusyTimes);
+      List<TimeRange> result = getAvailableTimes(optionalBusyTimes, request.getDuration());
+ 
+      // If optional attendee scheduling conflicts with mandatory scheduling, schedule
+      // only for mandatory attendees 
+      if(result.isEmpty() && !request.getAttendees().isEmpty()) {
+        return mandatoryResult;
+      }  
+      return result;
+    }
+    return mandatoryResult;
   }
 
-  // gets busy times for all request attendees
-  private List<TimeRange> getBusyTimes(Collection<Event> events, MeetingRequest request) {
+  // Gets busy times for all request attendees depending on if there are optional attendees
+  private List<TimeRange> getBusyTimes(Collection<Event> events, MeetingRequest request, String requestType, List<TimeRange> toAppend) {
```

Comment:
Maybe use a boolean for requestType, isMandatory.

Reply:
yes that is much better thank you!